src/ophyd_async/core/_hdf_dataset.py

```diff
@@ -19,6 +19,7 @@ class HDFDataset:
     dtype_numpy: str = ""
     multiplier: int = 1
     swmr: bool = False
+    chunk_size: int = None
```

Comment:
Maybe?
```suggestion
    chunk_size: int | None= None
```
or `Optional[None]`?